core-bundle/src/DataContainer/Palette.php

```diff
@@ -0,0 +1,100 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Contao\CoreBundle\DataContainer;
+
+use Contao\StringUtil;
+
+class Palette implements \Stringable
+{
+    /**
+     * @param array<string, array{fields: array<string>, hide: bool}> $config
+     */
+    private function __construct(private array $config = [])
+    {
+    }
+
+    public function __toString(): string
+    {
+        return $this->toString();
+    }
+
+    /**
+     * @return array<string, array{fields: array<string>, hide: bool}>
+     */
+    public function getConfig(): array
+    {
+        return $this->config;
+    }
+
+    public static function fromString(string $palette): self
+    {
+        return new self(self::explode($palette));
+    }
+
+    public function hasLegend(string $legend): bool
+    {
+        return isset($this->config[$legend]);
+    }
+
+    public function toString(): string
```

Comment:
So now that there is a `__toString()` method, we don't need this one anymore, do we?

Reply:
https://github.com/contao/contao/pull/8302#issuecomment-2865795598